mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -63,7 +63,7 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
 
   /** @group setParam */
   @Since("1.4.0")
-  def setSplits(value: Array[Double]): this.type = set(splits, value)
+  def setSplits(value: Array[Double]): this.type = set(splits, value.filter(!_.isNaN))
```

Comment:
rejecting? do you mean we throw an error for splits with NaN ?
User might not specify NaN bounded splits, but it could be set by other function calls, such as approxQuantile, from a dataset with quite a number of NaN